pkg/core/storelimit/sliding_window.go

```diff
@@ -46,7 +46,10 @@ func NewSlidingWindows(cap float64) *SlidingWindows {
 
 // Reset resets the capacity of the sliding windows.
 // It doesn't clear all the used, only set the capacity.
-func (s *SlidingWindows) Reset(cap float64, _ Type) {
+func (s *SlidingWindows) Reset(cap float64, typ Type) {
+	if typ != SendSnapshot {
```

Comment:
I don't know exactly what the next implementation looks like. Can we unify `SlidingWindows` and `limit` into a interface?  Introduce `Ack` for `StoreRateLimit` and `limit`.

Reply:
yes, I will expose the ack in the later pr.